src/Microsoft.AspNetCore.Razor/RazorTemplateEngine.cs

```diff
@@ -363,6 +368,50 @@ public GeneratorResults GenerateCode(TextReader input, CancellationToken? cancel
                 throw new ArgumentNullException(nameof(input));
             }
 
+            if (!Host.DesignTimeMode)
+            {
+                // During runtime we want code generation explosions to flow up into the calling code. At design time
+                // we want to capture these exceptions to prevent IDEs from crashing.
+                return GenerateCodeCore(input, className, rootNamespace, sourceFileName, checksum);
+            }
+
+            try
+            {
+                return GenerateCodeCore(input, className, rootNamespace, sourceFileName, checksum);
+            }
+            catch (Exception ex)
```

Comment:
`catch (Exception ex) when (Host.DesignTimeMode)` is an alternative in case you don't want to duplicate the call.